pennylane/transforms/batch_transform.py

```diff
@@ -158,9 +159,60 @@ def __init__(self, transform_fn, expand_fn=None, differentiable=True):
         self.transform_fn = transform_fn
         self.expand_fn = expand_fn
         self.differentiable = differentiable
+        self.qnode_wrapper = self.default_qnode_wrapper
         functools.update_wrapper(self, transform_fn)
 
-    def qnode_execution_wrapper(self, qnode, targs, tkwargs):
+    def custom_qnode_wrapper(self, fn):
+        """Register a custom QNode execution wrapper function
+        for the batch transform.
+
+        **Example**
+
+        .. code-block:: python
+
+            def my_transform(tape, *targs, **tkwargs):
+                ...
+                return tapes, processing_fn
+
+            @my_transform.custom_qnode_wrapper

```

Comment:
I don't really understand what this new wrapper is for (or how this decorator syntax is working). Is this related to the PR description part below?

> Finally, a small change was made to qml.batch_execute to allow transforms to override the default QNode execution logic (if needed). This need arises in the new QNode.

Reply:
Yes, exactly 🙂 Currently, `@qml.batch_transform` wraps a decorates a function that returns `tapes, processing_fn`, and abstracts away _everything_ else; constructing the QNode, execution, etc.

However I came across issues where certain batch transformations might need to override/define their own execution.

And then I ran into a difficulty with the abstraction -- since `@qml.batch_transform` can be used as a decorator, we can't simply rely on subclassing and method overriding. Instead, we need a solution similar to JAX/Autograd, where the transform itself can register replacement methods:

![image](https://user-images.githubusercontent.com/2959003/132868522-3394a27e-6bb7-49d4-92ea-4614d3ede2f9.png)

Reply:
Ahh, the example in the screenshot is very helpful, thanks!